edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/publisher/LogAnalyticsUpload.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.Diagnostics.Publisher
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.Devices.Edge.Util.AzureLogAnalytics;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+
+    public sealed class LogAnalyticsUpload : IMetricsPublisher
+    {
+        static readonly ILogger Log = Logger.Factory.CreateLogger<LogAnalyticsUpload>();
+        readonly string workspaceId;
+        readonly string workspaceKey;
+        readonly string logType;
+        readonly Guid guid;
+
+        public LogAnalyticsUpload(string workspaceId, string workspaceKey, string logType, Guid guid)
+        {
+            this.workspaceId = Preconditions.CheckNonWhiteSpace(workspaceId, nameof(workspaceId));
+            this.workspaceKey = Preconditions.CheckNonWhiteSpace(workspaceKey, nameof(workspaceKey));
+            this.logType = Preconditions.CheckNonWhiteSpace(logType, nameof(logType));
+            this.guid = Preconditions.CheckNotNull(guid);
```

Comment:
Guid cannot be null; only need to check it is not equal to Guid.Empty.

Reply:
I think I'm missing something. Can't any input be null?